collector/kernel/main.cc

```diff
@@ -563,111 +564,124 @@ int main(int argc, char *argv[]) {
-      throw std::runtime_error("Cannot start rate limit timer.");
+      res = uv_timer_start(&refill_log_rate_limit_timer, refill_log_rate_limit_cb,
+                           /*1s*/ 1000, /*1s*/ 1000);
+      if (res != 0) {
+        throw std::runtime_error("Cannot start rate limit timer.");
+      }
     }
-  }
-
-  /* initialize curl engine */
-  auto curl_engine = CurlEngine::create(&loop);
 
+    /* initialize curl engine */
+    auto curl_engine = CurlEngine::create(&loop);
 
-  auto maybe_proxy_config = config::HttpProxyConfig::read_from_env();
-  auto proxy_config = maybe_proxy_config ? &*maybe_proxy_config : nullptr;
-  AuthzFetcher authz_fetcher{*curl_engine, *authz_server, agent_key, agent_id, proxy_config};
 
-  auto intake_config = intake_config_handler.read_config(authz_fetcher.token()->intake());
+    auto maybe_proxy_config = config::HttpProxyConfig::read_from_env();
+    auto proxy_config = maybe_proxy_config ? &*maybe_proxy_config : nullptr;
+    AuthzFetcher authz_fetcher{*curl_engine, *authz_server, agent_key, agent_id, proxy_config};
 
-  if (disable_intake_tls.given()) {
-    intake_config.disable_tls(*disable_intake_tls);
-  }
-
-  // Initialize our kernel telemetry collector
-  KernelCollector kernel_collector{
-    bpf_src, intake_config, boot_time_adjustment,
-    aws_metadata.try_value(), gcp_metadata.try_value(),
-    configuration_data.labels(), loop, *curl_engine,
-    authz_fetcher,
-    enable_http_metrics,
-    enable_userland_tcp,
-    CgroupHandler::CgroupSettings{
-      .force_docker_metadata = *force_docker_metadata,
-      .dump_docker_metadata = *dump_docker_metadata,
-    },
-    nullptr,
-    bpf_dump_file.Get(),
-    HostInfo{
-      .os = OperatingSystem::Linux,
-      .os_flavor = integer_value(*host_distro),
-      .os_version = gcp_metadata ? gcp_metadata->image() : "unknown",
-      .kernel_headers_source = *kernel_headers_source,
-      .kernel_version = unamebuf.release,
-      .hostname = hostname
-    }, *entrypoint_error
-  };
+    auto intake_config = intake_config_handler.read_config(authz_fetcher.token()->intake());
 
-  authz_fetcher.auto_refresh(
+    if (disable_intake_tls.given()) {
+      intake_config.disable_tls(*disable_intake_tls);
+    }
 
-    loop,
-    std::bind(&KernelCollector::update_authz_token, &kernel_collector, std::placeholders::_1)
-  );
+    // Initialize our kernel telemetry collector
+    KernelCollector kernel_collector{
+      bpf_src, intake_config, boot_time_adjustment,
+        aws_metadata.try_value(), gcp_metadata.try_value(),
+        configuration_data.labels(), loop, *curl_engine,
+        authz_fetcher,
+        enable_http_metrics,
+        enable_userland_tcp,
+        CgroupHandler::CgroupSettings{
+        .force_docker_metadata = *force_docker_metadata,
+          .dump_docker_metadata = *dump_docker_metadata,
+          },
+        nullptr,
+          bpf_dump_file.Get(),
+          host_info,
+          *entrypoint_error
+                 };
```

Comment:
This should be clang-formatted.

Reply:
Yes, that is pretty ugly.  I clang-formatted these lines, but not the entire files changed because there are too many changes there.